File: frontend/src/types/comment.ts
export interface Comment {
    id: string;
    text: string;
    createdAt: string;
    username: string;
    avatarUrl?: string | null;
    userId: string;
    videoId: string;
    parentCommentId?: string | null;
    channelId?: string | null;
    repliesCount: number;
    likesCount: number;
    dislikesCount: number;
    isLiked: boolean;
    isDisliked: boolean;
    replies?: Comment[];
}

export interface CreateCommentRequest {
    videoId: string;
    text: string;
    parentCommentId?: string | null;
}

export interface CreateCommentResponse {
    message: string;
    comment: Comment;
}

File: backend/VidFlow/Models/Comment.cs
using System.ComponentModel.DataAnnotations;

namespace VidFlow.Models
{
    public class Comment
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [MaxLength(5000)]
        public string Text { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        public Guid UserId { get; set; }
        public User User { get; set; } = null!;

        public Guid VideoId { get; set; }
        public Video Video { get; set; } = null!;

        public Guid? ParentCommentId { get; set; }
        public Comment? ParentComment { get; set; }
        public ICollection<Comment> Replies { get; set; } = new List<Comment>();

        public int LikesCount { get; set; } = 0;
        public int DislikesCount { get; set; } = 0;
        public ICollection<CommentLike> Likes { get; set; } = new List<CommentLike>();
    }
}
